packages/vite/src/node/plugins/css.ts

```diff
@@ -610,6 +610,50 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
         )
       }
 
+      const inlineCSS = async () => {
+        // legacy build and inline css
+
+        // Entry chunk CSS will be collected into `chunk.viteMetadata.importedCss`
+        // and injected later by the `'vite:build-html'` plugin into the `index.html`
+        // so it will be duplicated. (https://github.com/vitejs/vite/issues/2062#issuecomment-782388010)
+        // But because entry chunk can be imported by dynamic import,
+        // we shouldn't remove the inlined CSS. (#10285)
+
+        chunkCSS = await finalizeCss(chunkCSS, true, config)
+        let cssString = JSON.stringify(chunkCSS)
+        cssString =
+          renderAssetUrlInJS(
+            this,
+            config,
+            chunk,
+            opts,
+            cssString,
+          )?.toString() || cssString
+        const style = `__vite_style__`
+        const injectCode =
+          `var ${style} = document.createElement('style');` +
+          `${style}.textContent = ${cssString};` +
+          `document.head.appendChild(${style});`
+        const wrapIdx = code.indexOf('System.register')
+        const executeFnStart =
+          code.indexOf('{', code.indexOf('execute:', wrapIdx)) + 1
+        const s = new MagicString(code)
+        s.appendRight(executeFnStart, injectCode)
+        if (config.build.sourcemap) {
+          // resolve public URL from CSS paths, we need to use absolute paths
+          return {
+            code: s.toString(),
+            map: s.generateMap({ hires: true }),
+          }
+        } else {
+          return { code: s.toString() }
+        }
+      }
+
+      if (!config.build.cssExtract) {
+        return await inlineCSS()
+      }
+
```

Comment:
Got build error in experimental component library:
```txt
$ pnpm run build

> my-library-vue-ts@0.0.1 build C:\Source\my-library-vue-ts
> vue-tsc && vite build

vite v4.4.0-beta.3 building for production...
✓ 9 modules transformed.
[vite:esbuild-transpile] Transform failed with 1 error:
my-library-vue-ts.es.js:1:12: ERROR: Expected "as" but found "__vite_style__"

Expected "as" but found "__vite_style__"
1  |  import {var __vite_style__ = document.createElement('style');__vite_style__.textContent = ".read-the-docs[data-v-8fe7e3eb]{color:#888}button[data-v-df630c6a]{padding:.6rem 1.25rem;background-color:#747bff;border:0}button[data-v-df630c6a]:hover{background-color:#535bf2}\n";document.head.appendChild(__vite_style__); defineComponent, ref, openBlock, createElementBlock, Fragment, createElementVNode, toDisplayString, pushScopeId, popScopeId, createTextVNode, reactive, normalizeClass, unref, renderSlot } from 'vue';    
   |              ^
2  |
3  |  const _withScopeId = (n) => (pushScopeId("data-v-8fe7e3eb"), n = n(), popScopeId(), n);


[vite:dts] Start generate declaration files...
✓ built in 263ms
[vite:dts] Declaration files built in 926ms.

error during build:
Error: Transform failed with 1 error:
my-library-vue-ts.es.js:1:12: ERROR: Expected "as" but found "__vite_style__"
    at failureErrorWithLog (C:\Source\my-library-vue-ts\node_modules\.pnpm\esbuild@0.18.20\node_modules\esbuild\lib\main.js:1649:15)
    at C:\Source\my-library-vue-ts\node_modules\.pnpm\esbuild@0.18.20\node_modules\esbuild\lib\main.js:847:29
    at responseCallbacks.<computed> (C:\Source\my-library-vue-ts\node_modules\.pnpm\esbuild@0.18.20\node_modules\esbuild\lib\main.js:703:9)
    at handleIncomingPacket (C:\Source\my-library-vue-ts\node_modules\.pnpm\esbuild@0.18.20\node_modules\esbuild\lib\main.js:762:9)
    at Socket.readFromStdout (C:\Source\my-library-vue-ts\node_modules\.pnpm\esbuild@0.18.20\node_modules\esbuild\lib\main.js:679:7)
    at Socket.emit (node:events:514:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
 ELIFECYCLE  Command failed with exit code 1.
```
### Reproducer
Project: https://github.com/IsraelDCastro/my-library-vue-ts
```sh
pnpm install
```
Replace installed `vite` by this PR's package.
In `vite.config.ts`, add:
```ts
 cssExtract: false,
```
Then build:
```sh
pnpm run build
```

Reply:
I can also confirm this is happening.

Reply:
@marcelobotega - in order to reproduce i just cloned your repo, built the resources in `packages/vite/` with `pnpm run build` then linked that package using `yarn link` and used it in my own library with `yarn link vite`.